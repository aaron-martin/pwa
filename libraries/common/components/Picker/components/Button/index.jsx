/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * The default button for the Picker component.
 * @returns {JSX} The button component.
 */
const PickerButton = ({ value, label, openList }) => {
  if (value !== null) {
    return (
      <button className={styles.button} onClick={openList}>
        <span className={styles.label}>{label}</span>
        <span className={styles.value}>{value}</span>
      </button>
    );
  }

  return (
    <button className={styles.button} onClick={openList}>
      <span className={styles.label}>{label}</span>
    </button>
  );
};

PickerButton.propTypes = {
  label: PropTypes.string.isRequired,
  openList: PropTypes.func.isRequired,
  value: PropTypes.string,
};

PickerButton.defaultProps = {
  value: null,
};

export default PickerButton;
