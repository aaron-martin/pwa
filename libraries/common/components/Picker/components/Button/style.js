/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

const button = css({
  display: 'block',
  width: '100%',
  textAlign: 'left',
  padding: '10px 20px',
}).toString();

const label = css({
  display: 'block',
}).toString();

const value = css({
  display: 'block',
  fontWeight: 'bold',
}).toString();

export default {
  button,
  label,
  value,
};
