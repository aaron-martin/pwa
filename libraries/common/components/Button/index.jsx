/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import style from './style';

/**
 * The button component.
 */
class Button extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
  };

  static defaultProps = {
    className: '',
    disabled: false,
    onClick: null,
  };

  /**
   * Getter for the calculated button props.
   * @returns {Object}
   */
  get buttonProps() {
    const buttonProps = {
      className: `${this.props.className} ${style}`,
      disabled: this.props.disabled,
      onClick: this.props.disabled ? null : this.props.onClick,
    };

    return buttonProps;
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    return (
      <button {...this.buttonProps}>
        {this.props.children}
      </button>
    );
  }
}

export default Button;
