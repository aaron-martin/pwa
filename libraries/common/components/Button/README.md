Button Component
---

The Button component provides basic functionality for rendering generic types of buttons. 
There are no styles defined for the Button, so this component should not be utilized directly. 
Instead, one of the Button derivates provided by the template library may be used. 

## Getting Started

```
import { Button } from 'Library/components';

<Button onClick={handleClick} />
```

## Props

### className

_Default_: `''`  
_Type_: `string`

Additional classes that are to be appended to the Picker container.

###### Usage:

```
<Button className="class1 class2" ... />
```

### onClick

_Type_: `Function`  

Accepts a callback function for the components click event.

### disabled

_Type_: `boolean`  
_Default_: `false`  

Enables or disable the button.

###### Usage:

```
<Button disabled ... />
```
---
