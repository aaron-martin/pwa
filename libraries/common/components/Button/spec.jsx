/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('<Button />', () => {
  it('should render the button', () => {
    const wrapper = shallow(<Button>My content</Button>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button').length).toBe(1);
  });

  it('should render the button in disabled state', () => {
    const wrapper = shallow(<Button disabled>My content</Button>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button[disabled]').length).toBe(1);
  });

  it('should trigger the click event', () => {
    const callback = jest.fn();

    const wrapper = shallow(<Button onClick={callback}>My content</Button>);

    wrapper.simulate('click');

    expect(wrapper).toMatchSnapshot();
    expect(callback).toHaveBeenCalled();
  });

  it('should not trigger the click event when disabled', () => {
    const callback = jest.fn();

    const wrapper = shallow(<Button disabled onClick={callback}>My content</Button>);

    wrapper.simulate('click');

    expect(wrapper).toMatchSnapshot();
    expect(callback).not.toHaveBeenCalled();
  });
});
