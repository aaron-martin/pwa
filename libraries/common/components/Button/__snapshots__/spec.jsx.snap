// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should not trigger the click event when disabled 1`] = `
<button
  className=" css-1rn65c0"
  disabled={true}
  onClick={null}
>
  My content
</button>
`;

exports[`<Button /> should render the button 1`] = `
<button
  className=" css-1rn65c0"
  disabled={false}
  onClick={null}
>
  My content
</button>
`;

exports[`<Button /> should render the button in disabled state 1`] = `
<button
  className=" css-1rn65c0"
  disabled={true}
  onClick={null}
>
  My content
</button>
`;

exports[`<Button /> should trigger the click event 1`] = `
<button
  className=" css-1rn65c0"
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
>
  My content
</button>
`;
